main.py

```diff
@@ -3,10 +3,16 @@
 import httpx
 import json
 import random
+from decimal import Decimal
+import threading
```

Comment:
do we need threading module?  we can achieve concurrency with asyncio

```
async def main():
    tasks = [task(f'Coroutine-{i}') for i in range(3)]
    await asyncio.gather(*tasks)
```

Reply:
Asyncio is better I suppose, whichever is more effective...

Reply:
Mainly the code already is on asyncio -- no need to introduce the other library :)